src/plugin/plugin_manager.js

```diff
@@ -174,10 +174,23 @@ export function getLatestVersionNumber() {
 }
 
 export function isUpdateAvailable() {
-  if (!pluginObject) return;
-  let current = pluginObject.VERSION;
-  let latest  = getLatestVersionNumber();
-  return utils.versionCompare(latest, current) > 0;
+  return new Promise ((resolve, reject) =>{
+    callWhenPluginReady(() => {
+      if (pluginObject) {
+        var current = pluginObject.VERSION;
+        var latest = getLatestVersionNumber();
+        utils.versionCompare(latest, current) > 0 ? resolve(true) : resolve(false);
+      } else {
+        reject("plugin object is not available");
+      }
+    });
+  });
+}
+
+export function checkForUpdate() {
```

Comment:
I don't think we need to keep this function.
At most, you could have a updateIfNewVersionAvailable() function, which would NOT depend on config.autoUpdate, and would just be 
if (isUpdateAvailable()) {
      updatePlugin();
    }